src/main/java/classify/textfilecode/TextFileParser.java

```diff
@@ -0,0 +1,194 @@
+     */
+    private static void parseTextFile(File[] fileList, Integer matchIndex, ArrayList <Student> masterStudentList){
+        String fileNameString = fileList[matchIndex].getName().trim();
+        String fullFilePath = INPUT_TEXT_FILE_DIRECTORY + "/" + fileNameString;
+        try {
+            FileReader textfileReader = new FileReader(fullFilePath);
+            BufferedReader line = new BufferedReader(textfileReader);
+            UI.println("Fetching the data from " + fileNameString + ".");
+            fetchDataFromTextFile(line, masterStudentList);
+
+        } catch (FileNotFoundException e) {
+            UI.println(ERROR_ACCESSING_THE_FILE);
+        }
+
+    }
+
+    /**
+     * Parse the students' subject name and classes attended before going over all the students taking this class
+     *
+     * @param line Every line of the file selected by user
+     * @param masterStudentList StudentList where students will be added.
+     */
+    private static void fetchDataFromTextFile(BufferedReader line, ArrayList <Student> masterStudentList) {
+        try{
+            String firstLine = line.readLine();
+            String [] subjectInfoStrings = firstLine.split("Subject:");
+            String subjectName = subjectInfoStrings[1].trim();
+
+            String secondLine = line.readLine();
+            String [] classInfoStrings = secondLine.split("Classes Attended:");
+            int totalClassesAttended = Integer.parseInt(classInfoStrings[1].trim());
+
+            while (line.ready()){
+                String input = line.readLine();
+                deconstructTextFile(input, subjectName, totalClassesAttended, masterStudentList);
+            }
+        } catch (IOException e) {
+            UI.println(ERROR_ACCESSING_THE_FILE);
+        }
+    }
+
+    /**
+     * Adds the new students into the student list and adds their subject, grade, and classes attended.
+     * Default value is used for gender, phone number, remarks.
+     *
+     * @param textFileInput Every line of the text file with the students' information
+     * @param subjectName Common subject of the students
+     * @param classesAttended Common number of classes attended
+     * @param masterStudentList StudentList where students will be added.
+     */
+    private static void deconstructTextFile (String textFileInput, String subjectName,
+                                        Integer classesAttended, ArrayList <Student> masterStudentList){
+        if (textFileInput.trim().isBlank()){
+            return;
+        }
+        String [] inputArr = textFileInput.split(MAIN_REGEX);
+
+
+        try{
+            //Set Name
+            Student student = new Student(AddStudent.splitName(inputArr[NAME].trim()));
+            InputParsing.checkForSpecialCharacters(student.getName());
+            //Set grades
+            double subjectGrades = Double.parseDouble(inputArr[GRADE_FOR_SUBJECT].trim());
```

Comment:
Might need to add the exception for when trying to read in a non number as a double. NumberFormatException I believe